sycl/include/CL/sycl/detail/pi.h

```diff
@@ -390,7 +390,8 @@ typedef enum {
   PI_COMMAND_TYPE_SVM_MEMCPY = CL_COMMAND_SVM_MEMCPY,
   PI_COMMAND_TYPE_SVM_MEMFILL = CL_COMMAND_SVM_MEMFILL,
   PI_COMMAND_TYPE_SVM_MAP = CL_COMMAND_SVM_MAP,
-  PI_COMMAND_TYPE_SVM_UNMAP = CL_COMMAND_SVM_UNMAP
+  PI_COMMAND_TYPE_SVM_UNMAP = CL_COMMAND_SVM_UNMAP,
+  PI_COMMAND_TYPE_MEMADVISE = 0x4207
```

Comment:
Do you absolutely need to add new true command type? Can't just use "fake" `PI_COMMAND_TYPE_USER`?

Reply:
If we are to extend the enumerations then we need to do it in a more robust (guarantee no overlap) and consistent (different backends need to handle those) way.

Reply:
I could do this yes, but I felt it was a good idea to do this as it is a defined value in intel's USM extension document and re-usable by the OpenCL backend. But if you'd rather we move away from that then that's totally fine with me, happy to replace the usage with `PI_COMMAND_TYPE_USER`.

Reply:
Let's stick to `PI_COMMAND_TYPE_USER` until we really need to distinguish. FWIW, this is also what Level-Zero plugin is using: https://github.com/intel/llvm/blob/0c0f4c5cd729fc9060fa4c7fd3ce5646cdf94f3e/sycl/plugins/level_zero/pi_level_zero.cpp#L5804

Reply:
Done